lib/request/flushGetRequest.js

```diff
@@ -21,42 +21,56 @@ module.exports = function flushGetRequest(request, listOfPaths, callback) {
-    // Since its a list of paths
-    var pathMap = request._pathMap;
-    var listKeys = Object.keys(listOfPaths);
-    var listIdx = 0, listLen = listKeys.length;
-    for (; listIdx < listLen; ++listIdx) {
-        var paths = listOfPaths[listIdx];
-        for (var j = 0, pathLen = paths.length; j < pathLen; ++j) {
-            var pathSet = paths[j];
-            var len = pathSet.length;
+    var model = request.requestQueue.model;
+    if (model._enablePathCollapse || model._enableRequestDeduplication) {
+        // Note on the if-condition: request deduplication uses request._pathMap,
+        // so we need to populate that field if the feature is enabled.
 
-            if (!pathMap[len]) {
-                pathMap[len] = [pathSet];
-            } else {
-                var pathSetsByLength = pathMap[len];
-                pathSetsByLength[pathSetsByLength.length] = pathSet;
+        // TODO: Move this to the collapse algorithm,
+        // TODO: we should have a collapse that returns the paths and
+        // TODO: the trees.
+
+        // Take all the paths and add them to the pathMap by length.
+        // Since its a list of paths
+        var pathMap = request._pathMap;
+        var listIdx = 0,
+            listLen = listOfPaths.length;
+        for (; listIdx < listLen; ++listIdx) {
+            var paths = listOfPaths[listIdx];
+            for (var j = 0, pathLen = paths.length; j < pathLen; ++j) {
+                var pathSet = paths[j];
+                var len = pathSet.length;
+
+                if (!pathMap[len]) {
+                    pathMap[len] = [pathSet];
+                } else {
+                    var pathSetsByLength = pathMap[len];
+                    pathSetsByLength[pathSetsByLength.length] = pathSet;
+                }
             }
         }
-    }
 
-    // now that we have them all by length, convert each to a tree.
-    var pathMapKeys = Object.keys(pathMap);
-    var pathMapIdx = 0, pathMapLen = pathMapKeys.length;
-    for (; pathMapIdx < pathMapLen; ++pathMapIdx) {
-        var pathMapKey = pathMapKeys[pathMapIdx];
-        pathMap[pathMapKey] = toTree(pathMap[pathMapKey]);
+        // now that we have them all by length, convert each to a tree.
+        var pathMapKeys = Object.keys(pathMap);
+        var pathMapIdx = 0,
+            pathMapLen = pathMapKeys.length;
+        for (; pathMapIdx < pathMapLen; ++pathMapIdx) {
+            var pathMapKey = pathMapKeys[pathMapIdx];
+            pathMap[pathMapKey] = toTree(pathMap[pathMapKey]);
+        }
     }
 
-    // Take the pathMapTree and create the collapsed paths and send those
-    // off to the server.
-    var collapsedPaths = request._collasped = toPaths(pathMap);
-    var jsonGraphData;
+    if (model._enablePathCollapse) {
+        // Take the pathMapTree and create the collapsed paths and send those
+        // off to the server.
+        requestPaths = toPaths(request._pathMap);
+    } else if (listOfPaths.length === 1) {
```

Comment:
what's the purpose behind differentiating between `listOfPaths` being single-element or multi-element array?

Reply:
Just a small optimization. When batch times are short or batching is disabled, or when request rates are low (relative to the batch timer), `listOfPaths` will contain only the single Array of path sets. Essentially, this is the happy path for the simple case.

Reply:
aligned offline. my point of confusion was that `requestPaths` seemed to be turned into different types in these conditional blocks. Jeroen explained how it's not. Would recommend adding a comment explaining the same thing to an inattentive observer.